taqo_cli/lib/experiment_service.dart

```diff
@@ -0,0 +1,249 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// @dart=2.9
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:logging/logging.dart';
+import 'package:taqo_common/model/event.dart';
+import 'package:taqo_common/model/experiment.dart';
+import 'package:taqo_common/net/paco_api.dart';
+import 'package:taqo_common/service/experiment_service_lite.dart';
+import 'package:taqo_common/storage/joined_experiments_storage.dart';
+import 'package:taqo_common/util/schedule_printer.dart' as schedule_printer;
+import 'package:taqo_common/util/zoned_date_time.dart';
+
+import 'experiment_paused_status_cache.dart';
+import 'platform_service.dart' as platform_service;
+
+final _logger = Logger('ExperimentService');
+
+class ExperimentService implements ExperimentServiceLite {
```

Comment:
Maybe mark all the duplicated classed from the flutter client with some searchable annotation like 'TODO refactor techdebt' or something so we are communicating what we are doing.

Reply:
Done